### DISCUSSION TITLE
Built-in Terminal

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#821] [Docs]
<meta url="https://github.com/johnlindquist/kit/discussions/821">
<meta id="D_kwDOEu7MBc4AP_up">
<meta sectionIndex="1">
<meta title="Built-in Terminal">
<meta section="Essentials">
<meta i="4">    
<meta path="docs/built-in-terminal">

## Use the Built-in Terminal

```js
// Name: Run Commands in the Terminal

import "@johnlindquist/kit"

await term({
  //defaults to home dir
  cwd: `~/.kenv`,
  command: `ls`,
  // The footer is optional. All terms continue with the same shortcuts
  footer: `ctrl+c or cmd+enter to continue`,
})
```

[Open run-commands-in-the-terminal in Script Kit](https://scriptkit.com/api/new?name=run-commands-in-the-terminal&url=https://gist.githubusercontent.com/johnlindquist/e1fbd791d67d772c047f2afcab087cff/raw/37f5ab92f8caafbf437d4f234818ebae67fb7fba/run-commands-in-the-terminal.js")

> The shell defaults to `zsh`. You can change your shell by setting the `KIT_SHELL` environment variable in the ~/kenv/.env, but most of the testing has been done with `zsh`.

**Comment by menzen** [comment]
Hi @johnlindquist 
great work !

is there a possibility to set the font in the built in terminal. I guess most of us get this screen here:

<img width="812" alt="Screenshot 2023-11-09 at 11 19 26" src="https://github.com/johnlindquist/kit/assets/1071108/f099ddf1-5d02-47b3-a96b-0507e657748b">

**Reply by johnlindquist** [reply]
@menzen If you're on the 2.0 release preview, in your `~/.kenv/.env`, you can add:

```
KIT_TERM_FONT=Geist Mono
```

**Reply by menzen** [reply]
Sorry, don't get it. The 2.0 release of ./kit ? its latest is v0.76.14 
and ./kenv i grabbed the latest from here https://github.com/johnlindquist/kenv 
but i had already the latest version.

Where is the 2.0 release preview? 

Thanks again for this awesome tools and your efforts !!!

**Reply by johnlindquist** [reply]
@menzen here:

https://github.com/johnlindquist/kit/discussions/1365

**Reply by menzen** [reply]
<img width="869" alt="Screenshot 2023-11-10 at 14 45 47" src="https://github.com/johnlindquist/kit/assets/1071108/a90c1a6c-4dab-4a7a-8e30-df94a8537f47">

Works, thanks a lot !
